src/avfoundation.cs

```diff
@@ -3715,6 +3784,91 @@ interface AVAssetTrack : NSCopying {
 		[Field ("AVAssetTrackTrackAssociationsDidChangeNotification")]
 		[Notification]
 		NSString TrackAssociationsDidChangeNotification { get; }
+
+		[Mac (10,10), NoiOS, NoTV, NoWatch]
+		[Export ("canProvideSampleCursors")]
+		bool CanProvideSampleCursors { get; }
+
+		[Mac (10,10), NoiOS, NoTV, NoWatch]
+		[return: NullAllowed]
+		[Export ("makeSampleCursorWithPresentationTimeStamp:")]
+		AVSampleCursor MakeSampleCursor (CMTime presentationTimeStamp);
+
+		[Mac (10,10), NoiOS, NoTV, NoWatch]
+		[return: NullAllowed]
+		[Export ("makeSampleCursorAtFirstSampleInDecodeOrder")]
+		AVSampleCursor MakeSampleCursorAtFirstSampleInDecodeOrder ();
+
+		[Mac (10,10), NoiOS, NoTV, NoWatch]
+		[return: NullAllowed]
+		[Export ("makeSampleCursorAtLastSampleInDecodeOrder")]
+		AVSampleCursor MakeSampleCursorAtLastSampleInDecodeOrder ();
+	}
+
+	[Mac (10,10), NoiOS, NoTV, NoWatch]
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject))]
+	interface AVSampleCursor : NSCopying {
+
+		[Export ("stepInDecodeOrderByCount:")]
+		long StepInDecodeOrder (long stepCount);
+
+		[Export ("stepInPresentationOrderByCount:")]
+		long StepInPresentationOrder (long stepCount);
+
+		[Export ("stepByDecodeTime:wasPinned:")]
+		CMTime StepByDecodeTime (CMTime deltaDecodeTime, out bool wasPinned);
+
+		[Export ("stepByPresentationTime:wasPinned:")]
+		CMTime StepByPresentationTime (CMTime deltaPresentationTime, out bool wasPinned);
+
+		[Export ("presentationTimeStamp")]
+		CMTime PresentationTimeStamp { get; }
+
+		[Export ("decodeTimeStamp")]
+		CMTime DecodeTimeStamp { get; }
+
+		[Export ("comparePositionInDecodeOrderWithPositionOfCursor:")]
+		NSComparisonResult ComparePositionInDecodeOrder (AVSampleCursor positionOfCursor);
+
+		[Export ("samplesWithEarlierDecodeTimeStampsMayHaveLaterPresentationTimeStampsThanCursor:")]
+		bool GetSamplesWithEarlierDecodeTimeStampsMayHaveLaterPresentationTimeStampsThan (AVSampleCursor positionOfCursor);
```

Comment:
I don't have a better name myself... but since it returns a `bool` it should be prefixed by `Has` and not `Get`. Reading it feels weird, maybe it should not have a prefix at all ? -> @chamons as an native English speaker